Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -111,7 +111,11 @@ class ChannelDTO: NSManagedObject {
         return load(by: request, context: context)
     }
     
-    static func loadOrCreate(cid: ChannelId, context: NSManagedObjectContext) -> ChannelDTO {
+    static func loadOrCreate(cid: ChannelId, context: NSManagedObjectContext, cache: PreWarmedCache?) -> ChannelDTO {
+        if let cachedObject = cache?.model(for: cid.rawValue, context: context, type: ChannelDTO.self) {
```

Comment:
With a caller like this, you don't need to pass `type` param to `model(for:context:type:)` func, since T will get resolved correctly

Reply:
Not unless you cast on the if let, which I want to avoid